lib/builders.js

```diff
@@ -83,56 +83,59 @@ class JSONBuilder {
   add(npmPackage) {
     const { name, version } = npmPackage;
 
-    const licenseText = npmPackage.text ? formatLicenseText(npmPackage.text) : '';
+    const licenseText = npmPackage.text
+      ? formatLicenseText(npmPackage.text)
+      : "";
 
     this.addCached(name, version, licenseText);
   }
 
-  buildLink(name, repository, licenseFile) {
-    return new Promise((resolve) => {
-      let licenseLink;
-      
-      if (licenseFile && repository) {
-        const licFileSegments = licenseFile.split('/');
-        const repoSegments = repository.split('/');
-        
-        licenseLink = `${repository}/blob/master/${licFileSegments[licFileSegments.length - 1]}`;
-        
-        request.get(licenseLink).then(
-          () => {
-            this.output[name].repository = licenseLink;
-            resolve(licenseLink);
-          },
-          (err) => {
-            console.log('bad', licenseLink);
-            resolve(null);
-          }
-        );
-      } else {
-        return resolve(null);
+  async buildLink(name, repository, licenseFile) {
+    if (licenseFile && repository) {
+      const licFileSegments = licenseFile.split("/");
+
+      const licenseLink = `${repository}/blob/master/${
+        licFileSegments[licFileSegments.length - 1]
+      }`;
+
+      try {
+        await axios.get(licenseLink);
+      } catch(err) {
+        console.log("bad", licenseLink);
+        return null;
       }
-    });
+
+      this.output[name].repository = licenseLink;
+      return licenseLink;
+    } else {
+      return null;
+    }
   }
 
   addEmpty(name, version) {
-    this.addCached(name, version, '');
+    this.addCached(name, version, "");
   }
 
   write() {
```

Comment:
How does this method get called? It does IO work here but doesn't return a callback or promise. Should something wait on completion?